src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/BackoffLogicManager.cs

```diff
@@ -21,19 +21,59 @@ internal class BackoffLogicManager : IDisposable
         private static readonly JavaScriptSerializer Serializer = new JavaScriptSerializer();
 
         private readonly TimeSpan defaultBackoffEnabledReportingInterval;
+        private readonly object lockConsecutiveErrors = new object();
 
         private TaskTimer pauseTimer = new TaskTimer { Delay = TimeSpan.FromSeconds(SlotDelayInSeconds) };
         private bool exponentialBackoffReported = false;
-        
+        private int consecutiveErrors;
+        private DateTimeOffset nextMinTimeToUpdateConsecutiveErrors = DateTimeOffset.MinValue;
+        private TimeSpan minIntervalToUpdateConsecutiveErrors;
+
         public BackoffLogicManager(TimeSpan defaultBackoffEnabledReportingInterval)
         {
             this.defaultBackoffEnabledReportingInterval = defaultBackoffEnabledReportingInterval;
+            this.minIntervalToUpdateConsecutiveErrors = TimeSpan.FromSeconds(SlotDelayInSeconds);
+        }
+
+        internal BackoffLogicManager(TimeSpan defaultBackoffEnabledReportingInterval, TimeSpan minIntervalToUpdateConsecutiveErrors) 
+            : this(defaultBackoffEnabledReportingInterval)
+        {
+            // This constructor is used from unit tests
+            this.minIntervalToUpdateConsecutiveErrors = minIntervalToUpdateConsecutiveErrors;
         }
 
         /// <summary>
         /// Gets or sets the number of consecutive errors SDK transmitter got so far while sending telemetry to backend.
         /// </summary>
-        public int ConsecutiveErrors { get; set; }
+        public int ConsecutiveErrors
+        {
+            get { return this.consecutiveErrors; }
+
+            set
```

Comment:
This code was moved form policies to manager without changing. I would not want to refactor it more since that would require rewriting all unit tests that we have now.